From 4cf388234017b34cb7644bc046a8030c367d6cb3 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:38:34 -0400
Subject: [ethernet] i40e: not all VSIs have rings

Message-id: <1409658042-22689-154-git-send-email-sassmann@redhat.com>
Patchwork-id: 90349
O-Subject: [RHEL7.1 PATCH 153/281] i40e: not all VSIs have rings
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Mitch Williams <mitch.a.williams@intel.com>

Once more, with feeling: not all VSIs have rings. To assume so is to
invite null pointers to your party.

Change-ID: I576858824468d9712d119fa1015a1f28c27712c4
Signed-off-by: Mitch Williams <mitch.a.williams@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit ddfda80f04f139fd147f1c517e79390daa157f27)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index c68d821..71b2f3f 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -657,7 +657,7 @@ static void i40e_update_link_xoff_rx(struct i40e_pf *pf)
  for (v = 0; v < pf->hw.func_caps.num_vsis; v++) {
   struct i40e_vsi *vsi = pf->vsi[v];
 
-  if (!vsi)
+  if (!vsi || !vsi->tx_rings[0])
    continue;
 
   for (i = 0; i < vsi->num_queue_pairs; i++) {
@@ -711,7 +711,7 @@ static void i40e_update_prio_xoff_rx(struct i40e_pf *pf)
  for (v = 0; v < pf->hw.func_caps.num_vsis; v++) {
   struct i40e_vsi *vsi = pf->vsi[v];
 
-  if (!vsi)
+  if (!vsi || !vsi->tx_rings[0])
    continue;
 
   for (i = 0; i < vsi->num_queue_pairs; i++) {
-- 
1.7.1